src/modules/gnd_pos_control/GroundRoverPositionControl.cpp

```diff
@@ -689,79 +423,57 @@ GroundRoverPositionControl::control_position(const math::Vector<2> &current_posi
-				|| (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF)) {
-			
-			 if (_launch_detection_state != LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS) {
-			 	_launch_detection_state = LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS;
-			 }
+			   || (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF)) {
+
+			if (_launch_detection_state != LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS) {
+				_launch_detection_state = LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS;
+			}
 
 			/* waypoint is a plain navigation waypoint or the takeoff waypoint, does not matter */
-			_gnd_control.navigate_waypoints(prev_wp, curr_wp, current_position, nav_speed_2d);
+			_gnd_control.navigate_waypoints(prev_wp, curr_wp, current_position, ground_speed_2d);
 			_att_sp.roll_body = _gnd_control.nav_roll();
+			_att_sp.pitch_body = 0.0f;
 			_att_sp.yaw_body = _gnd_control.nav_bearing();
-
-			/* Apply control output */
-			tecs_update_throttle(pos_sp_triplet.current.alt, calculate_target_speed(mission_target_speed), eas2tas,
-						   math::radians(_parameters.pitch_limit_min), math::radians(_parameters.pitch_limit_max),
-						   _parameters.throttle_min, _parameters.throttle_max, mission_throttle,
-						   math::radians(_parameters.pitch_limit_min), _global_pos.alt, ground_speed);
+			_att_sp.thrust = mission_throttle;
 
 		} else if (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
 
 			/* waypoint is a loiter waypoint so we want to stop*/
 			_gnd_control.navigate_loiter(curr_wp, current_position, pos_sp_triplet.current.loiter_radius,
-						    pos_sp_triplet.current.loiter_direction, nav_speed_2d);
+						     pos_sp_triplet.current.loiter_direction, ground_speed_2d);
+
 			_att_sp.roll_body = _gnd_control.nav_roll();
+			_att_sp.pitch_body = 0.0f;
 			_att_sp.yaw_body = _gnd_control.nav_bearing();
-
-			float alt_sp = pos_sp_triplet.current.alt;
-
-			tecs_update_throttle(alt_sp,
-						   calculate_target_speed(mission_target_speed),
-						   eas2tas,
-						   math::radians(_parameters.pitch_limit_min),
-						   math::radians(_parameters.pitch_limit_max),
-						   0, // _parameters.throttle_min,
-						   0, // _parameters.throttle_max,
-						   0, // _parameters.throttle_cruise,
-						   math::radians(_parameters.pitch_limit_min),
-						   _global_pos.alt,
-						   ground_speed);
+			_att_sp.thrust = mission_throttle;
 		}
 
 		if (was_circle_mode && !_gnd_control.circle_mode()) {
 			/* just kicked out of loiter, reset integrals */
 			_att_sp.yaw_reset_integral = true;
 		}
 
-	}else {
+	} else {
 		_control_mode_current = UGV_POSCTRL_MODE_OTHER;
 
+		_att_sp.roll_body = 0.0f;
+		_att_sp.pitch_body = 0.0f;
+		_att_sp.yaw_body = 0.0f;
+		_att_sp.thrust = 0.0f;
+
 		/* do not publish the setpoint */
 		setpoint = false;
 	}
 
-	/* Copy thrust output for publication */
-	if (_vehicle_status.engine_failure ||
```

Comment:
It's a FW feature and needs to be rethought anyway. In FW if it detects the throttle isn't working it's supposed to do a safe spiral descent.